source/write/delete.txt

```diff
@@ -0,0 +1,197 @@
+   :dedent:
+
+Modify the Delete Operation
+---------------------------
+
+You can modify the behavior of the ``MongoDB\Collection::deleteOne()`` and
+``MongoDB\Collection::deleteMany()`` methods by by passing an array that
+specifies option values as a parameter. The following table describes the
+options you can set in the array:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Option
+     - Description
+
+   * - ``collation``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the {+mdb-server+} manual.
+
+   * - ``writeConcern``
+     - | Sets the write concern for the operation.
+         For more information, see :manual:`Write Concern </reference/write-concern/>`
+         in the {+mdb-server+} manual.
+
+   * - ``hint``
+     - | Gets or sets the index to scan for documents. 
+         For more information, see the :manual:`hint statement </reference/command/delete/#std-label-deletes-array-hint>`
+         in the {+mdb-server+} manual.
+
+   * - ``let``
+     - | Specifies a document with a list of values to improve operation readability.
+         Values must be constant or closed expressions that don't reference document
+         fields. For more information, see the :manual:`let statement
+         </reference/command/delete/#std-label-delete-let-syntax>` in the
+         {+mdb-server+} manual.
+
+   * - ``session``
+     - | Specifies the client session to associate with the operation. For more
+         information, see :manual:`Session </reference/method/Session/>` in the
+         {+mdb-server+} manual.
+
+   * - ``comment``
+     - | Attaches a comment to the operation. For more information, see the :manual:`delete command
+         fields </reference/command/delete/#command-fields>` guide in the
+         {+mdb-server+} manual.
+
+Example
+~~~~~~~
+
+The following example calls the ``deleteMany()`` method to delete all documents in
+the ``restaurants`` collection that have a ``name`` value containing the string ``'Mongo'``.
+It also sets the ``comment`` option in an array parameter to add a comment to the
+operation:
+
+.. literalinclude:: /includes/write/delete.php
+   :start-after: start-delete-options
+   :end-before: end-delete-options
+   :language: php
+   :dedent:
+
+.. note::
+
+   If you replace the ``deleteMany()`` method with ``deleteOne()`` in
+   the preceding example, the library deletes only the first document that has
+   a ``name`` value containing ``'Mongo'``.
+
+Return Value
+------------
+
+The ``MongoDB\Collection::deleteOne()`` and ``MongoDB\Collection::deleteMany()``
+methods return a ``MongoDB\DeleteResult`` object. This class contains the
+following member functions:
+
+- ``isAcknowledged()``, which returns a boolean indicating whether the operation
+  was acknowledged
+- ``getDeletedCount()``, which returns the number of documents deleted
```

Comment:
I'm not sure if this is something you want to discuss here, but the getter methods throw if the write was not acknowledged, since the server wouldn't have returned any information and thus the driver has nothing to report. An exception (no pun intended) to this rule is getter methods for inserted IDs (on `insertOne()` and `insertMany()`) can always be called since those are collected client-side before we talk to the server.

Generally, most users don't run unacknowledged writes. And even if they did, there's probably little reason they'd call `isAcknowledged()` since they'd already know how they're executing the delete operation.

Probably TMI, but just wanted to provide context on how the methods on these result objects relate to each other. This isn't really discussed elsewhere beyond the API docs (e.g. [`DeleteResult::getDeletedCount()`](https://www.mongodb.com/docs/php-library/current/reference/method/MongoDBDeleteResult-getDeletedCount/).

Reply:
Added that the getDeletedCount() method throws an error if the write was not acknowledged.